### ISSUE TITLE
ShapRFECV extend support for regression and multiclass

### ISSUE THREAD

**Issue by Matgrb** [ing-bank/probatus#169]
**Problem Description**
Currently ShapRFECV supports only binary classification. It most probably works fine for other types of models.

**Desired Outcome**
1. The ShapRFECV and the EarlyStoppingShapRFECV fully support multiclass and regression models.
2. Code includes unit tests for multiclass and regression models.
3. Docs of the feature are updated to include support
4. Warnings are not shown for ShapRFECV and the EarlyStoppingShapRFECV during validating y labels in case they are not binary

**Comment by lproth** [comment]
I've implemented this for a project I'm working on. It works for LightGBM multi-class classification and regression. I haven't tested it on other models but I don't expect there would be any issues. I will make a pull request if you want to review it?

**Comment by Matgrb** [comment]
That would be great!

One request, if you make unit tests for lightgbm please add the following decorator to make sure that these tests are not executed on mac OS, since we currently have a problem with it.
```
@pytest.mark.skipif(os.environ.get("SKIP_LIGHTGBM") == "true", reason="LightGBM tests disabled")
```

**Comment by ReinierKoops** [comment]
fixed in #248
